src/main/java/scmspain/karyon/restrouter/core/ParamTypeResolver.java

```diff
@@ -16,11 +16,11 @@ public T resolveValueType(Class<T> classType, String value){
     } else if (classType.isAssignableFrom(boolean.class) || classType.isAssignableFrom(Boolean.class)) {
       return (T)Boolean.valueOf(value);
     } else if (classType.isAssignableFrom(long.class) || classType.isAssignableFrom(Long.class)) {
-      return (T)Boolean.valueOf(value);
+      return (T)Long.valueOf(value);
     } else if (classType.isAssignableFrom(short.class) || classType.isAssignableFrom(Short.class)) {
-      return (T)Boolean.valueOf(value);
+      return (T)Short.valueOf(value);
     } else if (classType.isAssignableFrom(float.class) || classType.isAssignableFrom(Float.class)) {
-      return (T)Boolean.valueOf(value);
+      return (T)Float.valueOf(value);
     }
     return null;
```

Comment:
@cristiangarciascmspain correct me if Im wrong but you are thinking in something like this:

http://symfony.com/doc/current/bundles/SensioFrameworkExtraBundle/annotations/converters.html